src/tools/replay/actors/ActorExecutionTrace.java

```diff
@@ -17,117 +18,127 @@
+      final TraceContextNode tracer) {
     TracingActivityThread t = getThread();
-    ((ActorTraceBuffer) t.getBuffer()).recordActorContext(actor, tracer);
+    ((ActorTraceBuffer) t.getBuffer()).recordActivityContext(activity, tracer);
   }
 
-  public static void recordSystemCall(final int dataId, final TraceActorContextNode tracer) {
+  public static void recordSystemCall(final int dataId, final TraceContextNode tracer) {
     TracingActivityThread t = getThread();
     ((ActorTraceBuffer) t.getBuffer()).recordSystemCall(dataId, tracer);
   }
 
-  public static void recordSystemCall(final int dataId, final TraceActorContextNode tracer,
+  public static void recordSystemCall(final int dataId, final TraceContextNode tracer,
       final TracingActivityThread t) {
     ((ActorTraceBuffer) t.getBuffer()).recordSystemCall(dataId, tracer);
   }
 
-  public static void intSystemCall(final int i, final TraceActorContextNode tracer) {
+  public static void intSystemCall(final int i, final TraceContextNode tracer) {
     ActorProcessingThread t = (ActorProcessingThread) getThread();
     TracingActor ta = (TracingActor) t.getCurrentActor();
     int dataId = ta.getDataId();
-    byte[] b = getExtDataByteBuffer(ta.getActorId(), dataId, Integer.BYTES);
+    byte[] b = getExtDataByteBuffer(ta.getId(), dataId, Integer.BYTES);
     TraceBuffer.UNSAFE.putInt(
         b, TraceBuffer.BYTE_ARR_BASE_OFFSET + EXT_DATA_HEADER_SIZE, i);
     recordSystemCall(dataId, tracer);
     t.addExternalData(b);
   }
 
-  public static void longSystemCall(final long l, final TraceActorContextNode tracer) {
+  public static void longSystemCall(final long l, final TraceContextNode tracer) {
     ActorProcessingThread t = (ActorProcessingThread) getThread();
     TracingActor ta = (TracingActor) t.getCurrentActor();
     int dataId = ta.getDataId();
-    byte[] b = getExtDataByteBuffer(ta.getActorId(), dataId, Long.BYTES);
+    byte[] b = getExtDataByteBuffer(ta.getId(), dataId, Long.BYTES);
     TraceBuffer.UNSAFE.putLong(
         b, TraceBuffer.BYTE_ARR_BASE_OFFSET + EXT_DATA_HEADER_SIZE, l);
     recordSystemCall(dataId, tracer);
     t.addExternalData(b);
   }
 
-  public static void doubleSystemCall(final double d, final TraceActorContextNode tracer) {
+  public static void doubleSystemCall(final double d, final TraceContextNode tracer) {
     ActorProcessingThread t = (ActorProcessingThread) getThread();
     TracingActor ta = (TracingActor) t.getCurrentActor();
     int dataId = ta.getDataId();
-    byte[] b = getExtDataByteBuffer(ta.getActorId(), dataId, Double.BYTES);
+    byte[] b = getExtDataByteBuffer(ta.getId(), dataId, Double.BYTES);
     TraceBuffer.UNSAFE.putDouble(
         b, TraceBuffer.BYTE_ARR_BASE_OFFSET + EXT_DATA_HEADER_SIZE, d);
     recordSystemCall(dataId, tracer);
     t.addExternalData(b);
   }
 
-  private static final int EXT_DATA_HEADER_SIZE = 3 * 4;
+  private static final int EXT_DATA_HEADER_SIZE = 16;
```

Comment:
Did you add another 4byte field to the header? should it be `4 * 4` then?

Reply:
one of the 3 integer fields was for actor id, which is now a long. hence 4 more byte needed. I'll change it to 2*Integer.Bytes + Long.Bytes

Reply:
Ah, yes, that would be even better!